server/src/main/java/chat/services/AuthorizationService.java

```diff
@@ -1,18 +1,31 @@
 package chat.services;
 
+import chat.database.UsersBase;
 import chat.model.User;
+import org.apache.log4j.Logger;
 
 import java.io.BufferedReader;
 import java.io.IOException;
 import java.io.PrintWriter;
+import java.sql.SQLException;
 import java.util.ArrayList;
 
 public class AuthorizationService {
     private static final String ENTER_NUMBER = "Enter the number '1' if you want to loggin or '2' if you want to registration: ";
-
+    private static final Logger LOGGER = Logger.getLogger(AuthorizationService.class);
     private ArrayList<User> clientBase = new ArrayList<>();
+    private UsersBase usersBase;
+
+    public User authorization(PrintWriter writer, BufferedReader reader) throws IOException, SQLException, ClassNotFoundException {
+
+        usersBase = new UsersBase();
+
+        clientBase.addAll(usersBase.dataBase());
```

Comment:
Here the same, you are getting all users from DB each time when someone authorizes. It is a good approach not to store authorized users in DB, as each call to DB is a high weight operation. You should leave some collection for "online users", but be sure that you have exactly one instance of your local storage (singleton).
Also, you should make HashMap, because it has constant algorithm complexity O(1) on writing and getting elements

Reply:
Ok, I used singleton but I left array list because i have one variable - User